fxrecorder/src/lib/retry.rs

```diff
@@ -0,0 +1,59 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::error::Error;
+use std::future::Future;
+use std::time::Duration;
+
+use derive_more::Display;
+use tokio::time::delay_for;
+
+#[derive(Debug, Display)]
+#[display(fmt = "failed after {} retries", retries)]
+/// An error that occurred when retrying a fallable operation.
+pub struct RetryError<E: Error + 'static> {
+    /// The last error that occurred.
+    source: E,
+
+    /// The number of retries.
+    retries: u32,
+}
+
+impl<E: Error + 'static> Error for RetryError<E> {
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        Some(&self.source)
+    }
+}
+
+/// Attempt to resolve the future returned by the given function `retries` times
+/// using exponential backoff between attempts.
+pub async fn exponential_retry<F, Fut, T, E>(
```

Comment:
The name kind of suggests to me that we will *immediately* execute `f`, and then retry `retries` times with an exponential backoff, rather than delaying first and then doing so. I think the behavior is what you want it to be though, right? So could we rename it something like `delayed_exponential_retry` to make it clear that it will delay first?